app/controllers/FaciaToolController.scala

```diff
@@ -32,10 +33,19 @@ class FaciaToolController(val config: ApplicationConfiguration, val acl: Acl, va
         Ok(Json.toJson(configJson)).as("application/json")}}}
 
   def getCollection(collectionId: String) = APIAuthAction.async { request =>
+    def f[A](x: Option[Future[A]])(implicit ec: ExecutionContext): Future[Option[A]] =
+      x match {
+        case Some(f) => f.map(Some(_))
+        case None    => Future.successful(None)
+      }
```

Comment:
In a few places I think you could replace this pattern match with 'Option.fold' which might be neater.